collect_app/src/main/java/org/odk/collect/android/receivers/NetworkReceiver.java

```diff
@@ -133,73 +143,99 @@ private void uploadForms(Context context, boolean isFormAutoSendOptionEnabled) {
 
             if (protocol.equals(context.getString(R.string.protocol_google_sheets))) {
-
-                if (PermissionUtils.checkIfGetAccountsPermissionGranted(context)) {
-                    GoogleAccountsManager accountsManager = new GoogleAccountsManager(Collect.getInstance());
-
-                    String googleUsername = accountsManager.getSelectedAccount();
-                    if (googleUsername.isEmpty()) {
-                        // just quit if there's no username
-                        running = false;
-                        return;
-                    }
-                    accountsManager.getCredential().setSelectedAccountName(googleUsername);
-                    instanceGoogleSheetsUploader = new InstanceGoogleSheetsUploader(accountsManager);
-                    instanceGoogleSheetsUploader.setUploaderListener(this);
-                    instanceGoogleSheetsUploader.execute(toSendArray);
-                } else {
-                    resultMessage = Collect.getInstance().getString(R.string.odk_permissions_fail);
-                    uploadingComplete(null);
-                }
+                sendInstancesToGoogleSheets(context, toSendArray);
             } else if (protocol.equals(context.getString(R.string.protocol_odk_default))) {
-                // get the username, password, and server from preferences
                 instanceServerUploader = new InstanceServerUploader();
                 instanceServerUploader.setUploaderListener(this);
-
                 instanceServerUploader.execute(toSendArray);
             }
         }
     }
 
+    private void sendInstancesToGoogleSheets(Context context, Long[] toSendArray) {
+        if (PermissionUtils.checkIfGetAccountsPermissionGranted(context)) {
+            GoogleAccountsManager accountsManager = new GoogleAccountsManager(Collect.getInstance());
+
+            String googleUsername = accountsManager.getSelectedAccount();
+            if (googleUsername.isEmpty()) {
+                // just quit if there's no username
+                running = false;
+                return;
+            }
+            accountsManager.getCredential().setSelectedAccountName(googleUsername);
+            instanceGoogleSheetsUploader = new InstanceGoogleSheetsUploader(accountsManager);
+            instanceGoogleSheetsUploader.setUploaderListener(this);
+            instanceGoogleSheetsUploader.execute(toSendArray);
+        } else {
+            resultMessage = Collect.getInstance().getString(R.string.odk_permissions_fail);
+            uploadingComplete(null);
+        }
+    }
+
+    /**
+     * Returns a list of longs representing the database ids of the instances that need to be
+     * autosent.
+     */
+    @NonNull
+    private List<Long> getInstancesToAutoSend(boolean isAutoSendAppSettingEnabled) {
+        List<Long> toUpload = new ArrayList<>();
+        Cursor c = new InstancesDao().getFinalizedInstancesCursor();
+
+        try {
```

Comment:
You can use [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) here.